Reviewed tables of built-in numeric types

pkulikov opened the pull request:
Along with the formatting updates, this PR:
- adds the reference to `BigInteger` API page to the [Integral Types Table](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/integral-types-table) article
- corrects (error occurs not in the assignment) and updates the [CS1021](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1021) article
- adds the `decimal` row to the [Floating-point types table](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/floating-point-types-table)

pkulikov committed:
Reviewed tables of built-in numeric types

mairaw commented:
It looks great to me, but I'd prefer if @BillWagner can give his final blessing. Thanks @pkulikov!

rpetrusha commented on docs/csharp/language-reference/keywords/floating-point-types-table.md:
```diff
@@ -1,26 +1,30 @@
 |----------|-----------------------|---------------|  
 |[float](float.md)|±1.5e−45 to ±3.4e38|7 digits|  
 |[double](double.md)|±5.0e−324 to ±1.7e308|15-16 digits|  
+|[decimal](decimal.md)|(-7.9 x 10<sup>28</sup> to 7.9 x 10<sup>28</sup>) / (10<sup>0</sup> to 10<sup>28</sup>)|28-29 digits|  
```
Unrelated to your PR, since the range is the same as in the [decimal topic](https://review.docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/decimal), but it's confusing. I think it's preferable to include a more simple form: -7.9228 x 10\<sup>28\<\sup> to 7.9228 x 10 \<sup>28\<\sup>, which is the value reported in the [Decimal](https://docs.microsoft.com/en-us/dotnet/api/system.decimal?view=netframework-4.7.2) documentation. If you make that change, it would also be good to change the range in the decimal keyword topic.

pkulikov replied:
@rpetrusha what do you think of providing a range in the format which is used for `float` and ` double`: from `+/- <closest to zero>` to `+/- <maximum absolute value>`: ±1 x 10<sup>-28</sup> to ±7.9228 x 10<sup>28</sup>

rpetrusha replied:
I think that that's fine, @pkulikov.

rpetrusha commented on docs/csharp/language-reference/keywords/integral-types-table.md:
```diff
@@ -1,36 +1,40 @@
+|[ulong](ulong.md)|0 to 18,446,744,073,709,551,615|Unsigned 64-bit integer|  
   
- If the value represented by an integer literal exceeds the range of `ulong`, a compilation error will occur.  
+ If the value represented by an integer literal exceeds <xref:System.UInt64.MaxValue?displayProperty=nameWithType>, a compile error [CS1021](../../misc/cs1021.md) occurs.
```
a compile error --> compiler error

rpetrusha commented on docs/csharp/language-reference/keywords/integral-types-table.md:
```diff
@@ -1,36 +1,40 @@
+|[ulong](ulong.md)|0 to 18,446,744,073,709,551,615|Unsigned 64-bit integer|  
   
- If the value represented by an integer literal exceeds the range of `ulong`, a compilation error will occur.  
+ If the value represented by an integer literal exceeds <xref:System.UInt64.MaxValue?displayProperty=nameWithType>, a compile error [CS1021](../../misc/cs1021.md) occurs.
```
It would be useful here to note that BigInteger has no practical upper limit -- something like, "To work around this issue, you can use the \<System.Numerics.BigInteger> type, which has no upper bound."

pkulikov replied:
I've moved the `BigInteger` mention from the **See also** section to a newly created **Remarks** section as a separate paragraph.

rpetrusha commented on docs/csharp/misc/cs1021.md:
````diff
@@ -1,30 +1,43 @@
+}
+```
+
+For information about how to instantiate a <xref:System.Numerics.BigInteger?displayProperty=nameWithType> instance whose value exceeds the range of the built-in numeric types, see the <xref:System.Numerics.BigInteger?displayProperty=nameWithType> reference page.
````
I think that the URL can link to the instantiating section: \<xref:System.Numerics.BigInteger#instantiating-a-biginteger-object?displayProperty=nameWithType>

rpetrusha reviewed (changes requested):
Thanks, @pkulikov, for these improvements. I've left a couple of comments and suggestions.

BillWagner reviewed (approved):
I like these changes @pkulikov 

I'll approve this PR, but I do agree with @rpetrusha 's comments.

pkulikov committed:
Addressed feedback

pkulikov commented:
@rpetrusha thank you for the review. I've addressed almost all of your feedback. I have a question about how to represent the range of the `decimal` type.

pkulikov committed:
Unified floating-point ranges

pkulikov commented:
@rpetrusha I've updated the ranges (also updated the `float` keyword article to be consistent with other floating-point topics). Please review the latest commits.

rpetrusha commented:
Thanks for all of the additional work on this PR, @pkulikov. It looks really good. I'll approve and merge now.
